Stripe/STPCustomer+Stripe_PaymentMethods.m

```diff
@@ -0,0 +1,85 @@
+//
+//  STPCustomer+Stripe_PaymentMethods.m
+//  Stripe
+//
+//  Created by Brian Dorfman on 3/17/17.
+//  Copyright © 2017 Stripe, Inc. All rights reserved.
+//
+
+#import "STPCustomer+Stripe_PaymentMethods.h"
+#import "STPPaymentConfiguration+Private.h"
+#import "STPPaymentMethodTuple.h"
+
+@implementation STPCustomer (Stripe_PaymentMethods)
+
+- (STPPaymentMethodTuple *)stp_paymentMethodTupleWithConfiguration:(STPPaymentConfiguration *)configuration {
+
+
+    NSOrderedSet<STPPaymentMethodType *> *availableTypes = configuration.availablePaymentMethodTypes;
+    // TODO: store last payment method locally or on stripe meta data so
+    // it can be a non-reusable source
+
+    id<STPPaymentMethod> selectedPaymentMethod = nil;
+    NSMutableArray<id<STPPaymentMethod>> *filteredSavedPaymentMethods = [NSMutableArray new];
+    for (id<STPSourceProtocol> source in self.sources) {
+        if ([source conformsToProtocol:@protocol(STPPaymentMethod)]) {
+            id<STPPaymentMethod> paymentMethod = (id<STPPaymentMethod>)source;
+            if ([availableTypes containsObject:paymentMethod.paymentMethodType]) {
+                [filteredSavedPaymentMethods addObject:paymentMethod];
+                if ([self.defaultSource.stripeID isEqualToString:source.stripeID]) {
+                    selectedPaymentMethod = paymentMethod;
+                }
+            }
+        }
+    }
+
+    NSArray<id<STPPaymentMethod>> *savedPaymentMethods = [self.class stp_sortedPaymentMethodsFromArray:filteredSavedPaymentMethods
+                                                                                       sortOrder:availableTypes];
+
+    NSMutableOrderedSet<STPPaymentMethodType *> *availablePaymentTypes = availableTypes.mutableCopy;
+    // If apple pay is misconfigured, remove it from the list
+    if (!configuration.applePayEnabled) {
+        [availablePaymentTypes removeObject:[STPPaymentMethodType applePay]];
+    }
+
+    // TODO: if other types are misconfigured, remove or warn? Also maybe need
```

Comment:
Yeah I'm not sure exactly where we should do this validation? Can't just do it when you set the available types, because you might do the setters in a different order (eg set available types first, then set redirect). Doing it right before showing the available options to the customer seemed best?